src/core/postNote.ts

```diff
@@ -0,0 +1,146 @@
+import joplin from 'api';
+import {Attachment} from '../model/attachment.model';
+import {EmailContent} from '../model/emailContent.model';
+import {EXPORT_TYPE} from '../constants';
+import {Attachments} from './postAttachments';
+import {ExportNoteBody} from '../model/exportNoteBdoy.model';
+import {AttachmentProperties} from 'src/model/attachmentProperties.mode';
+
+export class PostNote {
+    emailHtmlBody: string;
+    subject: string;
+    attachments: Attachment[];
+    folders: string[] = [];
+    tags: string[] = [];
+    noteBody: ExportNoteBody = {title: '', parent_id: '', body: '', body_html: '', markup_language: 1};
+
+    async createFolder(folder: string): Promise<void> {
+        await joplin.data.post(['folders'], null, {title: folder});
+    }
+
+    async createTag(tag: string): Promise<void> {
+        await joplin.data.post(['tags'], null, {title: tag});
+    }
+
+    async addFolders(folders: string[]): Promise<void> {
+        this.folders = folders;
+        const joplinFolders = await joplin.data.get(['folders']);
+
+        // Search by folder title
+        const joplinFoldersTitles = joplinFolders.items.map((joplinFolder)=> joplinFolder.title);
+
+        const newFolders = folders.filter((folder)=> !joplinFoldersTitles.includes(folder));
+
+        newFolders.forEach(async (folder)=>{
+            await this.createFolder(folder);
+        });
+    }
+
+    async addTags(tags: string[]): Promise<void> {
+        this.tags = tags;
+        const joplinTags = await joplin.data.get(['tags']);
+
+        // Search by tag title
+        const joplinTagsTitles = joplinTags.items.map((joplinTag)=> joplinTag.title);
+
+        const newTags = tags.filter((tag)=> !joplinTagsTitles.includes(tag));
+
+        newTags.forEach(async (tag)=>{
+            await this.createTag(tag);
+        });
+    }
+
+    async post(emailContent: EmailContent, folders: string[], tags: string []): Promise<void> {
+        if (folders.length === 0) return;
+
+        this.attachments = emailContent.attachments;
+        this.subject = emailContent.subject;
+
+        // to get only the HTML body of the email and ignore unnecessary data.
+        this.emailHtmlBody = new DOMParser().parseFromString(emailContent.html, 'text/html').body.innerHTML;
+
+        // Converting email to (html, markdown, text) depends on what export type is in the plugin settings.
+        await this.converter();
+
+        await this.addFolders(folders);
+        await this.addTags(tags);
+
+        const joplinFolders = await joplin.data.get(['folders']);
+        const joplinTags = await joplin.data.get(['tags']);
+
+        const emailFolders = joplinFolders.items.filter((e)=>{
+            return this.folders.includes(e.title);
+        });
+
+        const emailTags = joplinTags.items.filter((e)=>{
+            return this.tags.includes(e.title);
+        });
+
+        for (let i = 0; i < emailFolders.length; i++) {
+            const folder = emailFolders[i];
```

Comment:
Do I read this correctly that it creates a copy of the email in multiple folders? Tbh, I struggle to imagine why this may be needed...

Reply:
I deleted it because we don't need it now, but this function takes an array of the folders that you want the note in and for each folder, creates the note and assigns it the tags.